cpp/include/cudf/io_functions_cpp.h

```diff
@@ -0,0 +1,29 @@
+
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory> 
+#include <arrow/io/interfaces.h>
+#include <arrow/io/file.h>
+
+//I can't add the file_handle to args because args has to be compiled
+//for cffi
+gdf_error read_csv_arrow(csv_read_arg *args,
+		std::shared_ptr<arrow::io::RandomAccessFile> file_handle);
```

Comment:
Instead of making it an arg why don't we just use it under the hood for implementation details? That way a user either provides a buffer or a string. If it's a string we treat it as a URI and feed it into the `arrow::io::...` call which handles the accessing and memory mapping. If it's a buffer we feed it into `arrow::io::BufferReader` to do a zero copy operation to then have an essentially standardized interface to the CSV memory.

I guess my point is why not make this an implementation detail for the general `read_csv` call?

Reply:
cc @mjsamoht @harrism @felipeblazing for their opinions because this seems silly to me